cosmos/cache.py

```diff
@@ -397,3 +404,49 @@ def create_cache_profile(version: str, profile_content: str) -> Path:
     profile_yml_path = profile_yml_dir / DEFAULT_PROFILES_FILE_NAME
     profile_yml_path.write_text(profile_content)
     return profile_yml_path
+
+
+def is_cache_lockfile_enabled() -> bool:
+    from dbt.version import __version__
```

Comment:
@pankajastro we may want to revisit this, since it won't work for a few use cases.

Although this will work for `ExecutionMode.LOCAL` usually, it will not work if users use `dbt_executable_path` to a `dbt` installation separate from Airflow (most Astro customers).

This will also not work for `ExecutionMode.VIRTUALENV`.

https://astronomer.github.io/astronomer-cosmos/getting_started/execution-modes.html

Reply:
Thank you for the pointer. Yes, you are right; I have removed dbt as a dependency and now checking only for the presence of the lock file in the project to enable it.